### PR TITLE
Add configured option for assets selection

### PR DESCRIPTION
**Pull request by eric3873**

### CODE REVIEW

File: include/fty_common_db_asset.h [hunk 1]
```diff
@@ -77,6 +77,10 @@ int select_asset_element_all_with_warranty_end(tntdb::Connection& conn, std::fun
 // returns 0 if succesful
 // returns -1 if error occurs
 
+int select_assets_by_container(tntdb::Connection& conn, uint32_t element_id, std::vector<uint16_t> types,
```
Comment (jeffdegott) [inline include/fty_common_db_asset.h:80]: add a comment explaining that api with configured arg.

File: src/fty_common_db_asset.cc [hunk 2]
```diff
@@ -734,6 +803,77 @@ int select_assets_all_container(tntdb::Connection& conn, std::vector<uint16_t> t
...
+                    "         AND EXISTS "
+                    "             (SELECT id_asset_device_dest FROM t_bios_asset_link_type "
+                    "             AS l JOIN t_bios_asset_link AS a ON a.id_asset_link_type=l.id_asset_link_type "
+                    "             WHERE name=\"power chain\" AND t.id_asset_element=a.id_asset_device_dest))) "
+                    // else for other device with no power chain, parent must be defined
+                    " OR (t.id_type IN (" + filter_type_id + ") "
+                    "     AND t.id_subtype NOT IN (" + filter_subtype_id + ") "
+                    "     AND t.id_parent IS NOT NULL)) ";
+            }
+        }
```
Comment (jeffdegott) [inline src/fty_common_db_asset.cc:875]: add else case to log error "configured malformed"

File: src/fty_common_db_asset.cc [hunk 3]
```diff
@@ -679,7 +748,7 @@ int select_assets_all_container(tntdb::Connection& conn, std::vector<uint16_t> t
             " FROM "
             "   t_bios_asset_element as t";
 
-        if (!subtypes.empty() || !types.empty() || status != "" || without != "") {
+        if (!subtypes.empty() || !types.empty() || status != "" || without != "" || !configured.empty()) {
```
Comment (jeffdegott) [inline src/fty_common_db_asset.cc:751]: if configured is non empty and not "yes" or "no"... the select request is malformed (the WHERE part is not set)?

File: src/fty_common_db_asset.cc [hunk 4]
```diff
@@ -433,6 +433,67 @@ int select_assets_by_container(tntdb::Connection& conn, uint32_t element_id, std
             }
         }
 
+        // if configured option is defined
+        if (!configured.empty()) {
```
Comment (jeffdegott) [inline src/fty_common_db_asset.cc:437]: test the value.  configured == "yes" or "no" instead

File: src/fty_common_db_asset.cc [hunk 5]
```diff
@@ -433,6 +433,67 @@ int select_assets_by_container(tntdb::Connection& conn, uint32_t element_id, std
...
+                    "         AND EXISTS "
+                    "             (SELECT id_asset_device_dest FROM t_bios_asset_link_type "
+                    "             AS l JOIN t_bios_asset_link AS a ON a.id_asset_link_type=l.id_asset_link_type "
+                    "             WHERE name=\"power chain\" AND v.id_asset_element=a.id_asset_device_dest))) "
+                    // else for other device with no power chain, parent must be defined
+                    " OR (v.id_type IN (" + filter_type_id + ") "
+                    "     AND v.id_asset_device_type NOT IN (" + filter_subtype_id + ") "
+                    "     AND v.id_parent1 IS NOT NULL)) ";
+            }
+        }
```
Comment (jeffdegott) [inline src/fty_common_db_asset.cc:495]: else error 'bad value'...

File: include/fty_common_db_asset.h (hunk 1 above)
Comment (eric3873) [reply include/fty_common_db_asset.h:80]: Take into account

File: src/fty_common_db_asset.cc (hunk 2 above)
Comment (eric3873) [reply src/fty_common_db_asset.cc:995]: Take into account

File: src/fty_common_db_asset.cc (hunk 4 above)
Comment (eric3873) [reply src/fty_common_db_asset.cc:437]: Take into account

File: src/fty_common_db_asset.cc (hunk 5 above)
Comment (eric3873) [reply src/fty_common_db_asset.cc:555]: Take into account

**Pull request merged**